src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.AspireWorkloadDeprecation.targets

```diff
@@ -0,0 +1,40 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.AspireWorkloadDeprecation.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved.
+***********************************************************************************************
+-->
+<Project>
+
+  <!--
+    This target detects old Aspire applications that still depend on the deprecated Aspire workload.
+
+    Starting with Aspire 9.0.0, Aspire moved from a workload-based distribution model to NuGet packages
+    with the Aspire.AppHost.Sdk. Projects using the old workload-based approach need to be upgraded.
+
+    Detection logic:
+    - IsAspireHost='true' indicates this is an Aspire host project
+    - AspireHostingSDKVersion is set by Aspire.AppHost.Sdk in new projects (9.0.0+)
+    - If AspireHostingSDKVersion is missing or < 9.0.0, this is likely an old workload-based project
```

Comment:
you say likely here. Are there any scenarios where we'd be giving customers a build error if we shouldn't?

Reply:
There is a very unlikely scenario where the new MSBuild SDK doesn't restore properly, and since that is the thing that defines AspireHostingSDKVersion you could run into a situation where the project is using the new  aspire but didn't detect this property (as the sdk failed to load/restore) so in that case it would be a false positive. Worth noting that while we've had reports of this happening once or twice in the past, we have never been able to repro it.